src/hooks/useSession.ts

```diff
@@ -0,0 +1,54 @@
+import { useEffect } from "react";
+import { useCookies } from "react-cookie";
+import { decodeToken, useJwt } from "react-jwt";
+import { toast } from "sonner";
+import { Session, User } from "@/types/session.types";
+import { COOKIE_AUTH_NAME, COOKIE_OPTIONS } from "@/lib/auth/constants";
+
+export default function useSession(): Session {
+    const [cookies, setCookie, removeCookie] = useCookies([COOKIE_AUTH_NAME]);
+    const { decodedToken, isExpired, reEvaluateToken } = useJwt<User>(
+        cookies[COOKIE_AUTH_NAME],
+    );
+
+    useEffect(() => {
+        const sessionToken = cookies[COOKIE_AUTH_NAME];
+        if (!sessionToken) {
+            return;
+        }
+
+        if (sessionToken && (!decodedToken || isExpired)) {
```

Comment:
The logic here is quite confusing. The following code get's executed in two scenarios
(a) The session token exists, but we can't decode it => The cookie is tampered with or corrupted somehow
(b) The cookie is expired
In both cases removing the original token and reevaluating it will not fix. You should send a request to the server to get a new token. Since this is not in the scope of this PR, consider writing a TODO comment.